boxsdk/util/api_call_decorator.py

```diff
@@ -0,0 +1,53 @@
+# coding: utf-8
+
+from __future__ import unicode_literals, absolute_import
+
+from functools import update_wrapper
+
+
+def api_call(func):
+    """
+    Designates the decorated method as one that makes a Box API call.
+    The decorated method can then accept a new keyword argument `request_data`,
+    a dictionary of key-value pairs to be passed to the network layer for API
+    calls made by the method.
+
+    :param func:
+        The method to decorate.
+    :type func:
+        `callable`
+    :return:
+        A wrapped method that can pass extra request data to the network layer.
+    :rtype:
+        :class:`APICallWrapper`
+    """
+    wrapper = APICallWrapper(func)
+    update_wrapper(wrapper, func)
+    return wrapper
+
+
+class APICallWrapper(object):
+
+    def __init__(self, func_that_makes_an_api_call):
+        super(APICallWrapper, self).__init__()
+        self._func_that_makes_an_api_call = func_that_makes_an_api_call
+
+    def __get__(self, _instance, owner):
+        def call(*args, **kwargs):
+            instance = _instance
+            if instance is None:
+                # If this is being called as an unbound method, the instance is the first arg.
+                if owner is not None and len(args) > 0 and isinstance(args[0], owner):
+                    instance = args[0]
+                    args = args[1:]
+                else:
+                    raise AttributeError
+            request_data = kwargs.pop('request_data', {})
```

Comment:
Did you want to keep this name, or use something else? Chris had suggested `network_parameters`.

Some other ideas: `network_arguments`, `extra_network_arguments`, `extra_request_arguments`, `extra_network_parameters`, `custom_network_parameters`, `custom_network_arguments`, `custom_request_arguments`, `custom_request_parameters`.

Also, in a future version, it might be cool to allow header customization in the same way. But we don't need that right now.